bin/improver-wxcode

```diff
@@ -0,0 +1,120 @@
+# ARE DISCLAIMED. IN NO EVENT SHALL THE COPYRIGHT HOLDER OR CONTRIBUTORS BE
+# LIABLE FOR ANY DIRECT, INDIRECT, INCIDENTAL, SPECIAL, EXEMPLARY, OR
+# CONSEQUENTIAL DAMAGES (INCLUDING, BUT NOT LIMITED TO, PROCUREMENT OF
+# SUBSTITUTE GOODS OR SERVICES; LOSS OF USE, DATA, OR PROFITS; OR BUSINESS
+# INTERRUPTION) HOWEVER CAUSED AND ON ANY THEORY OF LIABILITY, WHETHER IN
+# CONTRACT, STRICT LIABILITY, OR TORT (INCLUDING NEGLIGENCE OR OTHERWISE)
+# ARISING IN ANY WAY OUT OF THE USE OF THIS SOFTWARE, EVEN IF ADVISED OF THE
+# POSSIBILITY OF SUCH DAMAGE.
+"""CLI to generate weather symbols."""
+
+import argparse
+import iris
+import numpy as np
+from argparse import RawTextHelpFormatter
+
+from improver.wxcode.weather_symbols import WeatherSymbols
+from improver.wxcode.wxcode_utilities import expand_nested_lists
+from improver.wxcode.wxcode_decision_tree import wxcode_decision_tree
+
+
+def interogate_decision_tree():
+    """
+    Obtain a list of necessary inputs from the decision tree as it is currently
+    defined. Return a list of the diagnostic names, the thresholds needed,
+    and whether they are thresholded above of below these values.
+
+    Returns:
+        list:
+            Returns a list of strings, an entry for each diagnostic required,
+            including threshold details.
+    """
+
+    # Get current weather symbol decision tree and populate a list of
+    # required inputs for printing.
+    queries = wxcode_decision_tree()
+    requirements = {}
+    relative = {}
+
+    for query in queries.itervalues():
+        diagnostics = expand_nested_lists(query, 'diagnostic_fields')
+        for index, diagnostic in enumerate(diagnostics):
+            if diagnostic not in requirements:
+                requirements[diagnostic] = []
+                relative[diagnostic] = []
+            requirements[diagnostic].extend(
+                expand_nested_lists(query, 'diagnostic_thresholds')[index])
+            relative[diagnostic].append(
+                expand_nested_lists(query, 'diagnostic_conditions')[index])
+
+    output = []
+    for requirement in requirements:
+        entries = np.array([entry for entry in requirements[requirement]])
+        relations = np.array([entry for entry in relative[requirement]])
+        _, thresholds = np.unique(np.array([item.points.item()
+                                            for item in entries]),
+                                  return_index=True)
+        output.append('{}; thresholds: {}'.format(
+                requirement, ', '.join([
+                        '{} {} ({})'.format(str(relation),
+                                            str(threshold.points.item()),
+                                            str(threshold.units))
+                        for threshold, relation in
+                        zip(entries[thresholds], relations[thresholds])])))
+    return output
+
+
+def main():
+    """Parser to accept input data and an output destination before invoking
```

Comment:
Do you think 'main()' should be at the top of the module above 'interrogate_decision_tree()'  - for consistency with layout of other CLI's ?